linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml

```diff
@@ -0,0 +1,44 @@
+<def-group>
+  <definition class="compliance" id="sudoers_validate_passwd" version="1">
+   {{{ oval_metadata("Ensure invoking user's password for privilege escalation when using sudo") }}}
+      <criteria operator="AND">
+      <criterion comment="Check Defaults !targetpw exists in /etc/sudoers file" test_ref="test_sudoers_targetpw_config" />
+      <criterion comment="Check Defaults !rootpw exists in /etc/sudoers file" test_ref="test_sudoers_rootpw_config" />
+      <criterion comment="Check Defaults !runaspw exists in /etc/sudoers file" test_ref="test_sudoers_runaspw_config" />
+      </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="Ensure invoking user's password for privilege escalation when using sudo"
+  id="test_sudoers_targetpw_config" version="1">
+    <ind:object object_ref="object_test_sudoers_targetpw_config" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="Ensure invoking user's password for privilege escalation when using sudo"
+  id="test_sudoers_rootpw_config" version="1">
+    <ind:object object_ref="object_test_sudoers_rootpw_config" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="Ensure invoking user's password for privilege escalation when using sudo"
+  id="test_sudoers_runaspw_config" version="1">
+    <ind:object object_ref="object_test_sudoers_runaspw_config" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_test_sudoers_targetpw_config" version="1">
+    <ind:filepath>/etc/sudoers</ind:filepath>
+    <ind:pattern operation="pattern match">^Defaults !targetpw$\r?\n</ind:pattern>
```

Comment:
can there be spaces at the begining? What is the reason for the \r and \n at the end?

Reply:
No there cannot be spaces in the beginning and \r matches carriage return and \n matches a new line.